corehq/apps/export/views.py

```diff
@@ -1927,18 +1973,36 @@ class CreateNewDailySavedFormExport(DailySavedExportMixin, CreateNewCustomFormEx
 
 class BaseEditNewCustomExportView(BaseModifyNewCustomView):
 
+    def dispatch(self, request, *args, **kwargs):
+        try:
+            new_export_instance = self.new_export_instance
+        except ResourceNotFound:
+            new_export_instance = None
+        if (
+            new_export_instance
+            and new_export_instance.sharing in [SharingOption.EXPORT_ONLY, SharingOption.PRIVATE]
+            and new_export_instance.owner_id != request.couch_user.user_id
+        ):
+            raise Http404
```

Comment:
Probably could use `get_document_or_404` here

Reply:
The logic is different here - we 404 if the user doesn't have the edit permissions for the export, rather than 404 if the export doesn't exist (that's handled via the existing code).

Reply:
👍